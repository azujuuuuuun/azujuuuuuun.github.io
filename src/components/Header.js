import React from 'react'
import { Link } from 'gatsby'

export default props => {
  const { isHome } = props
  return (
    <header>
      {isHome ? (
        <h1>
          <Link to="/">azujuuuuuun.github.io</Link>
        </h1>
      ) : (
        <Link to="/">azujuuuuuun.github.io</Link>
      )}
    </header>
  )
}
