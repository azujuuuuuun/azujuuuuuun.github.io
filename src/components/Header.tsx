import React from 'react';
import { Link } from 'gatsby';

import locales from '../locales';

interface Props {
  isHome?: boolean;
}

const Header: React.VFC<Props> = props => {
  const { isHome } = props;
  return (
    <header>
      {isHome ? (
        <h1>
          <Link to="/">{locales.siteName}</Link>
        </h1>
      ) : (
        <Link to="/">{locales.siteName}</Link>
      )}
    </header>
  );
};

export default Header;
