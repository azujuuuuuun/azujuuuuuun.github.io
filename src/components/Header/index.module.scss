.Header {
  &__title {
    font-size: 3.2rem;
    font-weight: bold;
    overflow-wrap: break-word;
  }
}
