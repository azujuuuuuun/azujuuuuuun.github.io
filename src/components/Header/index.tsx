import React from "react";

const Header: React.FC = () => {
  return (
    <header>
      <h1
        style={{
          fontSize: "3.2rem",
          fontWeight: "bold",
          overflowWrap: "break-word",
        }}
      >
        azujuuuuuun.github.io
      </h1>
    </header>
  );
};

export default Header;
