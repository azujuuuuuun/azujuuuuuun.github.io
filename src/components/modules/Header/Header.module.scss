@use "../../../styles/variables" as *;

.Header {
  display: block;
  padding: $space-xs $space-l;

  &__title {
    font-size: $font-size-xxl;
    font-weight: bold;
    overflow-wrap: break-word;
  }
}
