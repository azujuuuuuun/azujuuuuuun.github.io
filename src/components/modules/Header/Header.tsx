import clsx from "clsx";
import type React from "react";

interface HeaderProps {
  className?: string;
}

export const Header: React.FC<HeaderProps> = ({ className }) => {
  return (
    <header className={clsx("block px-l py-xs", className)}>
      <h1 className="text-xxl font-bold break-words">azujuuuuuun.github.io</h1>
    </header>
  );
};
