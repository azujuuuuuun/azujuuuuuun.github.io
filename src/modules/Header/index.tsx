import React from "react";
import clsx from "clsx";
import styles from "./index.module.scss";

interface HeaderProps {
  className?: string;
}

export const Header: React.FC<HeaderProps> = ({ className }) => {
  return (
    <header className={clsx(styles.Header, className)}>
      <h1 className={styles.Header__title}>azujuuuuuun.github.io</h1>
    </header>
  );
};
