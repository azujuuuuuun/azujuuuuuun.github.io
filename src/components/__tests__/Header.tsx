import React from 'react';
import renderer from 'react-test-renderer';

import Header from '../Header';

describe('Header', (): void => {
  it('renders correctly', (): void => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
