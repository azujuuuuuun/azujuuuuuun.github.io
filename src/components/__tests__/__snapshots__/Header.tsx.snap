// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header>
  <a
    aria-current="page"
    className=""
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
    style={Object {}}
  >
    azujuuuuuun.github.io
  </a>
</header>
`;
